src/pytorch_lightning/overrides/fairscale.py

```diff
@@ -14,20 +14,16 @@
 from typing import Optional, Union
 
 import torch.nn as nn
-from lightning_utilities.core.imports import module_available
 
 import pytorch_lightning as pl
+from lightning_lite.strategies.fairscale import _FAIRSCALE_AVAILABLE
 from pytorch_lightning.overrides.base import (
     _LightningModuleWrapperBase,
     _LightningPrecisionModuleWrapperBase,
     unwrap_lightning_module,
 )
-from pytorch_lightning.utilities.imports import _IS_WINDOWS
 from pytorch_lightning.utilities.rank_zero import rank_zero_deprecation
 
-_FAIRSCALE_AVAILABLE = not _IS_WINDOWS and module_available("fairscale.nn")
```

Comment:
well this os for meta package creation

Reply:
Ok, I'm not sure actually I understand what's going on. I don't see how fairscale has anything to do with the meta package creation, but indeed the build fails so this is strange.

Reply:
The challenge here is the whole class is declared inside the `if` condition...
Inside the meta, we replace add func and class declaration with imports so as `_FAIRSCALE_AVAILABLE` is imported in the PL module, it is not imported meta module, so it is missing in the module... the solution is to:
1. keep `_FAIRSCALE_AVAILABLE` declaration even if it is duplicate
2. declare the class outside `if` conditions which would need some more refactor

Reply:
Thank you, now it makes more sense! 

> declare the class outside if conditions which would need some more refactor

I will try this first, seems like the easiest solution.